src/main/kotlin/no/nav/familie/ef/sak/repository/domain/AndelTilkjentYtelse.kt

```diff
@@ -9,8 +12,61 @@ data class AndelTilkjentYtelse(@Column("belop")
                                val stønadFom: LocalDate,
                                @Column("stonad_tom")
                                val stønadTom: LocalDate,
-                               val type: YtelseType)
+                               @Column("person_ident")
+                               val personIdent: String,
+                               @Column("periode_id")
+                               val periodeId: Long? = null,
+                               @Column("forrige_periode_id")
+                               val forrigePeriodeId: Long? = null,
+                               val type: YtelseType) {
+
+    private fun erTilsvarendeForUtbetaling(other: AndelTilkjentYtelse): Boolean {
+        return (this.personIdent == other.personIdent
+                && this.stønadFom == other.stønadFom
+                && this.stønadTom == other.stønadTom
+                && this.beløp == other.beløp
+                && this.type == other.type)
+    }
+
+    fun erNull() = this.beløp == 0
+
+    companion object {
+
+        /**
+         * Merk at det søkes snitt på visse attributter (erTilsvarendeForUtbetaling)
+         * og man kun returnerer objekter fra receiver (ikke other)
+         */
+        fun Set<AndelTilkjentYtelse>.snittAndeler(other: Set<AndelTilkjentYtelse>): Set<AndelTilkjentYtelse> {
+            val andelerKunIDenne = this.subtractAndeler(other)
+            return this.subtractAndeler(andelerKunIDenne)
+        }
+
+        fun Set<AndelTilkjentYtelse>.disjunkteAndeler(other: Set<AndelTilkjentYtelse>): Set<AndelTilkjentYtelse> {
+            val andelerKunIDenne = this.subtractAndeler(other)
+            val andelerKunIAnnen = other.subtractAndeler(this)
+            return andelerKunIDenne.union(andelerKunIAnnen)
+        }
+
+        fun nullAndel(kjedeId: KjedeId, periodeId: PeriodeId) =
+                kjedeId.tilNullAndelTilkjentYtelse(periodeId)
+
+        private fun Set<AndelTilkjentYtelse>.subtractAndeler(other: Set<AndelTilkjentYtelse>): Set<AndelTilkjentYtelse> {
+            return this.filter { a ->
+                other.none { b -> a.erTilsvarendeForUtbetaling(b) }
+            }.toSet()
+        }
+    }
+
+}
 
 enum class YtelseType(val klassifisering: String) {
-    OVERGANGSSTØNAD("??")
+    OVERGANGSSTØNAD("EFOG");
```

Comment:
Kunne vi brukt FAGSASKSYSTEM som er definert i Økonomiklient slik at koden EFOG ikke må endres flere steder hvi det skulle bli aktuelt?

Reply:
Dette er egentlig to forskjellige ting: Vi har både klassekode på utbetalingsperiodene vi sender, og vi setter fagsystem på toppnivå i oppdraget. Vet ikke om de alltid er 1-1 for ef-ytelsene, @nav-jorn-ola-birkeland? I så fall er jeg enig i at vi kan legge det på ett og samme sted.

Reply:
Ja, det er litt tilfeldig at det er det samme nå, så det vil være litt villedende å peke på det samme. MEN jeg tenker det er litt dumt å binde klassifiseringen til enum'en, så jeg flytter ut den.